azurerm/internal/services/monitor/resource_arm_monitor_action_rule.go

```diff
@@ -0,0 +1,591 @@
+				ValidateFunc:  validate.ActionGroupID,
+				ConflictsWith: []string{"suppression"},
+			},
+
+			"suppression": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				MaxItems:      1,
+				ConflictsWith: []string{"action_group_id"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"recurrence_type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(alertsmanagement.Always),
+								string(alertsmanagement.Once),
+								string(alertsmanagement.Daily),
+								string(alertsmanagement.Weekly),
+								string(alertsmanagement.Monthly),
+							}, false),
+						},
+
+						"schedule": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"end_date": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validate.ActionRuleScheduleDate,
+									},
+
+									"end_time": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validate.ActionRuleScheduleTime,
+									},
+
+									"start_date": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validate.ActionRuleScheduleDate,
+									},
+
+									"start_time": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validate.ActionRuleScheduleTime,
+									},
+
+									"recurrence_weekly": {
+										Type:          schema.TypeSet,
+										Optional:      true,
+										MinItems:      1,
+										ConflictsWith: []string{"suppression.0.schedule.0.recurrence_monthly"},
+										Elem: &schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validation.StringInSlice(weekDays, false),
+										},
+									},
+
+									"recurrence_monthly": {
+										Type:          schema.TypeSet,
+										Optional:      true,
+										ConflictsWith: []string{"suppression.0.schedule.0.recurrence_weekly"},
+										Elem: &schema.Schema{
+											Type:         schema.TypeInt,
+											ValidateFunc: validation.IntBetween(1, 31),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"condition": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"alert_context": schemaActionRuleAlertContextCondtion(),
+
+						"alert_rule_id": schemaActionRuleAlertRuleIDCondtion(),
+
+						"description": schemaActionRuleDescriptionCondtion(),
+
+						"monitor": schemaActionRuleMonitorCondtion(),
+
+						"monitor_service": schemaActionRuleMonitorServiceCondtion(),
+
+						"severity": schemaActionRuleSeverityCondtion(),
+
+						"target_resource_type": schemaActionRuleTargetResourceTypeCondtion(),
```

Comment:
this is really hard to understand like this. Given that almost all of these are only used once could we unwrap the call the the underlying  `schemaActionRule` function so there is more context for those reading this?